src/main/java/org/zalando/nakadi/repository/MultiTimelineEventConsumer.java

```diff
@@ -0,0 +1,265 @@
+package org.zalando.nakadi.repository;
+
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.zalando.nakadi.domain.ConsumedEvent;
+import org.zalando.nakadi.domain.EventTypePartition;
+import org.zalando.nakadi.domain.NakadiCursor;
+import org.zalando.nakadi.domain.PartitionStatistics;
+import org.zalando.nakadi.domain.Timeline;
+import org.zalando.nakadi.domain.TopicPartition;
+import org.zalando.nakadi.exceptions.InvalidCursorException;
+import org.zalando.nakadi.exceptions.NakadiException;
+import org.zalando.nakadi.exceptions.NakadiRuntimeException;
+import org.zalando.nakadi.exceptions.ServiceUnavailableException;
+import org.zalando.nakadi.service.timeline.TimelineService;
+import org.zalando.nakadi.service.timeline.TimelineSync;
+import org.zalando.nakadi.util.NakadiUtils;
+
+public class MultiTimelineEventConsumer implements EventConsumer {
+    private final String clientId;
+    private final Map<EventTypePartition, NakadiCursor> latestOffsets = new HashMap<>();
+    private final Map<String, List<Timeline>> consumedEventTypes = new HashMap<>();
+    private final Map<String, TimelineSync.ListenerRegistration> timelineRefreshListeners = new HashMap<>();
+    private final Map<TopicRepository, EventConsumer> eventConsumers = new HashMap<>();
+    private final LinkedList<ConsumedEvent> eventsQueue = new LinkedList<>();
+    private final Map<EventTypePartition, String> borderOffsets = new HashMap<>();
+    private final TimelineService timelineService;
+    private final TimelineSync timelineSync;
+    private final BlockingQueue<Runnable> queuedTasks = new LinkedBlockingQueue<>();
+    private static final Logger LOG = LoggerFactory.getLogger(MultiTimelineEventConsumer.class);
+
+    public MultiTimelineEventConsumer(
+            final String clientId,
+            final TimelineService timelineService,
+            final TimelineSync timelineSync) {
+        this.clientId = clientId;
+        this.timelineService = timelineService;
+        this.timelineSync = timelineSync;
+    }
+
+    @Override
+    public Set<TopicPartition> getAssignment() {
+        return latestOffsets.values().stream().map(NakadiCursor::getTopicPartition).collect(Collectors.toSet());
+    }
+
+
+    @Override
+    public Optional<ConsumedEvent> readEvent() {
```

Comment:
what is the expected time of execution of this method? Can it me more than 1 second? I ask this because EventStream class which uses it, has some expectations. 
For example if batch_flush_timeout = 1, then EventStream want to send batch (or empty batch) at least every second. And if this method will run more than that - that can produce behavior which user doesn't expect.